web_client/components/containers/analysis.jsx

```diff
@@ -148,6 +150,28 @@ const AnalysisContainer = connect(
       .then(() => dispatch(actions.openFileSelectorDialog()))
     ),
 
+    onItemSave: (item, name, itemWidget) => {
+      dispatch(actions.ensurePrivateDirectory({ returnModel: true }))
+        .then((parent) => {
+          let data = {
+            folderId: parent.id,
+            name
+          };
+
+          data = Object.entries(data)
```

Comment:
There may be a library function that does this. If it exists, it will probably handle corner cases better than this homebrew function.